src/memory_provider.c

```diff
@@ -80,17 +84,20 @@ umf_result_t umfMemoryProviderFree(umf_memory_provider_handle_t hProvider,
 void umfMemoryProviderGetLastNativeError(umf_memory_provider_handle_t hProvider,
                                          const char **ppMessage,
                                          int32_t *pError) {
+    ASSERT(hProvider != NULL);
```

Comment:
UMF_CHECK?

Reply:
I cannot use it here since this function returns void.